snap/validate.go

```diff
@@ -170,3 +177,73 @@ func ValidateApp(app *AppInfo) error {
 	}
 	return nil
 }
+
+// ValidatePathVariables ensures that given path contains only $SNAP, $SNAP_DATA or $SNAP_COMMON.
```

Comment:
interesting that we don't support having a `$` that _isn't_ interpolating a variable (i.e., no `\$`)